content-services/content-sources-backend
# Fixes 182: change api errors to error objects

Adds error object that returns on API errors to satisfy [this standard](https://jsonapi.org/format/#errors). It looks like:
```
errors: [
   {
      status: 400,
      title: "error creating repo",
      detail: "url already exists"
   }
]
```

Also specifies error object in openAPI spec for 400,404,500 for all endpoints.

This replaces the default echo error handler with `customHTTPErrorHandler` in `pkg/config/config.go`. It also moves custom errors (`DaoError` and `HandlerError`) to a package named `errors`. This way they can be imported by everything, including `pkg/config.go`

TODO

- [x] Tests (I'll fix the tests once direction of PR is okay)
- [x] Return most common error code as error code for bulk create. See "[generically applicable](https://jsonapi.org/format/#errors-processing)".

To test:

This changes the error response for all API errors, so cause errors and check that the response is as expected. Also check the API spec has been updated to correctly reflect the changes.
